[InfinityTwo] iP

InfinityTwo opened the pull request:
### Infinity Bot
> A rather lame bot, I might add

_Features_

- GUI
- ...and that's about it really, nothing special here

**How to use?**

1. Go to the [Github page](https://github.com/InfinityTwo/ip/) of my fork
2. Download the latest release
3. Run the file 😄 

**Which file is the main file for developing and how do I run it?**
Find the file under "src/main/java/infinity/ui/Launcher.java" and use this constructor.
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(GraphicalUserInterface.class, args);
    }
}
```

Checklist for what's left

- [x] It works
- [x] Test cases updated
- [ ] ???

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

InfinityTwo committed:
Rename to Infinity, Greet and Exit for bot

InfinityTwo committed:
Remove .class files tracking

InfinityTwo committed:
Modify README.md, removing Duke

InfinityTwo committed:
Customise bot to Echo

InfinityTwo committed:
Add tasklist with adding

InfinityTwo committed:
Add Mark as Done

InfinityTwo committed:
Add Todos, Events, Deadlines

InfinityTwo committed:
Add Exceptions

InfinityTwo committed:
Add text testing

InfinityTwo committed:
Add deletable tasks

InfinityTwo committed:
Change to 4 tabs indentation to follow styling

InfinityTwo committed:
Add enums

InfinityTwo committed:
Add save and load to save file functionality

InfinityTwo committed:
Merge branch 'branch-Level-7'

InfinityTwo committed:
Add proper support for Date & Time

InfinityTwo committed:
Merge branch 'branch-Level-8'

InfinityTwo committed:
Modify classes for better OOP

InfinityTwo committed:
Separate into packages for bot

InfinityTwo committed:
Merge remote-tracking branch 'origin/add-gradle-support'

InfinityTwo committed:
Add basic Gradle support

InfinityTwo committed:
Add JUnit tests for Parser and TaskList

InfinityTwo committed:
Release snapshot 03w01a

InfinityTwo committed:
Add find feature to bot

InfinityTwo committed:
Add javadoc throughout

InfinityTwo committed:
Add coding standards

InfinityTwo committed:
Modify to fit coding standards for Javadoc

InfinityTwo committed:
Merge branch 'branch-A-JavaDoc'

InfinityTwo committed:
Fix merge conflicts between A-Javadoc and A-CodingStandard branches

InfinityTwo committed:
Fix merge conflicts between master and branch-Level-9

InfinityTwo committed:
Move try catch for save to Storange.java

InfinityTwo committed:
Update Infinity.java to allow autosave on change

InfinityTwo committed:
Fix checkstyle

InfinityTwo committed:
Merge branch 'branch-A-CheckStyle'

InfinityTwo committed:
Add basic JavaFX, parts 1 to 3

InfinityTwo committed:
Add part 4 of JavaFX and Move bot to GUI based

InfinityTwo committed:
Update JavaDoc

InfinityTwo committed:
Merge branch 'branch-Level-10'

chiangqinkang commented on src/main/java/infinity/command/Command.java:
```diff
@@ -0,0 +1,100 @@
+     * @return The bot output.
+     * @throws InfinityException If there are errors encountered parsing or executing commands.
+     */
+    public static String findCommand(String currentInput, TaskList botTasks) throws InfinityException {
```
I feel that this could be clearer if a switch case statement was used instead. Also consider having a secondary function to help with bounds checking, this would better help with information hiding and lead to cleaner code.

chiangqinkang commented on src/main/java/infinity/parser/Parser.java:
```diff
@@ -0,0 +1,102 @@
+     * @param maxNo The maximum value of the number.
+     * @return True if the value is within the specified range and is a number, false otherwise.
+     */
+    private static boolean isNumberInRange(String value, int min, int max, int minNo, int maxNo) {
```
Perhaps you could split this into multiple lines for clarity, rather than having one long return statement?

chiangqinkang reviewed (commented):
Good, comprehensive test case coverage, even added comments for private methods (helpful for understanding code). Could clean up certain parts for better code legibility.

zlch commented on src/main/java/infinity/Infinity.java:
```diff
@@ -0,0 +1,58 @@
+import infinity.ui.Ui;
+import infinity.ui.components.DialogBox;
+import javafx.scene.image.Image;
+import javafx.scene.layout.VBox;
```
I like how the packages are separated nicely.

zlch commented on src/main/java/infinity/command/Command.java:
```diff
@@ -0,0 +1,100 @@
+
+            botOutput = Storage.saveFile(botTasks.getTasks());
+
+        } else if (currentInput.equals(KnownCommands.HELP.toString().toLowerCase())) {
```
Could the toString() and toLowercase() be combined to a single function to increase readability.

zlch reviewed (commented):
LGTM. Just some nits.

InfinityTwo committed:
Add some various assert lines

InfinityTwo committed:
Modify bot code for better code quality

InfinityTwo committed:
Add missing assert for switch

InfinityTwo committed:
Add CI

InfinityTwo committed:
Merge pull request #1 from InfinityTwo/branch-A-Assertions

A-Assertions

InfinityTwo committed:
Fix A-Assertion conflicts

InfinityTwo committed:
Modify Github CI

InfinityTwo committed:
Merge pull request #2 from InfinityTwo/branch-A-CodeQuality

A-CodeQuality

InfinityTwo committed:
Add Mass Operation to Bot

Mass Operations will work for some commands.
This includes deadline, delete, event, find,
mark and todo. We use && to delimit each operation.
To use it, an example is mark 1 && 2 && 3

Operations will not work for multicommands like
delete 1 && 2 && find abc

InfinityTwo committed:
Merge branch 'branch-BCD-Extension'

InfinityTwo committed:
Update TaskList and removed one indentation

InfinityTwo committed:
Modify README

InfinityTwo committed:
Merge branch 'master' of https://github.com/InfinityTwo/ip

InfinityTwo committed:
Add Final Build for Week 6

Add Contributors of myself
Add User Guide
Add Title to UI
Add UI Image
Fix final minor styling issues

InfinityTwo committed:
Modify testcases and Add more testcases to new methods

InfinityTwo committed:
Rename index.html for README homepage

InfinityTwo committed:
Modify Readme for website

InfinityTwo committed:
Fix default directory

InfinityTwo committed:
Change a unicode char to > instead
